presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestTpchCostBasedPlan.java

```diff
@@ -0,0 +1,102 @@
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tpch.ColumnNaming;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.facebook.presto.SystemSessionProperties.JOIN_DISTRIBUTION_TYPE;
+import static com.facebook.presto.SystemSessionProperties.JOIN_REORDERING_STRATEGY;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchConnectorFactory.TPCH_COLUMN_NAMING_PROPERTY;
+import static java.lang.String.format;
+
+/**
+ * This class tests cost-based optimization rules related to joins. It contains unmodified TPCH queries.
+ * This class is using TPCH connector configured in way to mock Hive connector with unpartitioned TPCH tables.
+ */
+public class TestTpchCostBasedPlan
+        extends BaseCostBasedPlanTest
+{
+    /*
+     * CAUTION: The expected plans here are not necessarily optimal yet. Their role is to prevent
+     * inadvertent regressions. A conscious improvement to the planner may require changing some
+     * of the expected plans, but any such change should be verified on an actual cluster with
+     * large amount of data.
+     */
+
+    public TestTpchCostBasedPlan()
+    {
+        super(() -> {
+            String catalog = "local";
+            SessionBuilder sessionBuilder = testSessionBuilder()
+                    .setCatalog(catalog)
+                    .setSchema("sf3000.0")
+                    .setSystemProperty("task_concurrency", "1") // these tests don't handle exchanges from local parallel
+                    .setSystemProperty(JOIN_REORDERING_STRATEGY, JoinReorderingStrategy.AUTOMATIC.name())
+                    .setSystemProperty(JOIN_DISTRIBUTION_TYPE, JoinDistributionType.AUTOMATIC.name());
+
+            LocalQueryRunner queryRunner = LocalQueryRunner.queryRunnerWithFakeNodeCountForStats(sessionBuilder.build(), 8);
+            queryRunner.createCatalog(
+                    catalog,
+                    new TpchConnectorFactory(1, false, false),
+                    ImmutableMap.of(TPCH_COLUMN_NAMING_PROPERTY, ColumnNaming.SIMPLIFIED.name()));
+            return queryRunner;
+        });
+    }
+
+    @Override
+    protected Stream<Query> getQueries()
+    {
+        return IntStream.rangeClosed(1, 22)
+                .mapToObj(i -> format("q%02d", i))
+                .map(queryId -> new Query(queryId, tpchQuery(queryId)));
+    }
+
+    private static Path tpchQuery(String queryId)
+    {
+        return Paths.get(format("./presto-benchto-benchmarks/src/main/resources/sql/presto/tpch/%s.sql", queryId));
+    }
+
+    @Override
+    protected Path getExpectedJoinOrderingFile(String queryId)
+    {
+        return Paths.get(format("./presto-main/src/test/resources/tpch-join-ordering/%s.txt", queryId));
```

Comment:
Ditto about path

Reply:
This is one is more tricker. It is used for generation. It is easy to read from resources, but it is difficult to update them.